app/controllers/api/transformation_mappings_controller.rb

```diff
@@ -2,9 +2,8 @@ module Api
   class TransformationMappingsController < BaseController
     def create_resource(_type, _id, data = {})
       raise "Must specify transformation_mapping_items" unless data["transformation_mapping_items"]
-      TransformationMapping.new(data.except("transformation_mapping_items")).tap do |mapping|
+      TransformationMapping.create(data.except("transformation_mapping_items")).tap do |mapping|
```

Comment:
`.create.tap` then `.save` will do two transactions instead of one.  Will this actually save the transformation_mapping_items if you don't save afterwards?

The .new.tap was weird, the best way to do this would be
```
tm = TransformationMapping.new()
tm.transformation_mapping_items = create_mapping_items
tm.save!
```

Reply:
@agrare, there is no .save  after the create statement.  Will there be an implicit .save call that you're referring to?  In my previous push, I had the following, which successfully created transformation mappings:
```
    def create_resource(_type, _id, data = {})
      raise "Must specify transformation_mapping_items" unless data["transformation_mapping_items"]
      TransformationMapping.create(data.except("transformation_mapping_items")).tap do |mapping|
        mapping.transformation_mapping_items = create_mapping_items(data["transformation_mapping_items"], mapping)
      end
    rescue StandardError => err
      raise BadRequestError, "Could not create Transformation Mapping - #{err}"
    end
```
If I use `TransformationMapping.new()`, then the following code does not work:
```
tm = TransformationMapping.new(data.except("transformation_mapping_items"))
tm.transformation_mapping_items = create_mapping_items
tm.save!
```
To make it work, I must use the following:
```
      mapping = TransformationMapping.new(data.except("transformation_mapping_items"))
      mapping.save!
      mapping.transformation_mapping_items = create_mapping_items(data["transformation_mapping_items"], mapping)
      mapping.save!
```
Based on last week discussion, it is not acceptable because if creating `TransformationMappingItem` fails validation, then the `TransformationMapping` does not get deleted.

Reply:
> If I use TransformationMapping.new(), then the following code does not work:

What part doesn't work?

Reply:
Using `TransformationMapping.new()` led to the failure of TransformationMappingItem validations, which blocked from creating transformation mapping items.  For a `TransformationMapping` to successfully create `transformation_mapping_items`, I used `TransformationMapping.create`.  The other option was to use `TransformationMapping.new()` with 2 `save!` statements.

Reply:
So right now the transformation_mapping_items validations will only work if the transformation_mapping is saved to the db, we are going to investigate in a follow-up but for now creating the transformation_mapping, then saving the transformation_mapping_items will work